tests/test_black.py

```diff
@@ -1832,6 +1833,42 @@ def test_read_pyproject_toml(self) -> None:
         self.assertEqual(config["exclude"], r"\.pyi?$")
         self.assertEqual(config["include"], r"\.py?$")
 
+    def test_read_pyproject_src_toml(self) -> None:
+        test_toml_file = THIS_DIR / "test_src.toml"
+        fake_ctx = FakeContext()
+        black.read_pyproject_toml(
+            fake_ctx, FakeParameter(), str(test_toml_file),
+        )
+        config = fake_ctx.default_map
+        self.assertEqual(config["verbose"], "1")
+        self.assertEqual(config["check"], "no")
+        self.assertEqual(config["diff"], "y")
+        self.assertEqual(config["color"], "True")
+        self.assertEqual(config["line_length"], "79")
+        self.assertEqual(config["target_version"], ["py36", "py37", "py38"])
+        self.assertEqual(config["exclude"], r"\.pyi?$")
+        self.assertEqual(config["include"], r"\.py?$")
+        self.assertEqual(config["src"], ["setup.py"])
+        self.assertEqual(fake_ctx.params.get("src"), ("setup.py",))
+
+    def test_read_pyproject_src_toml_with_src_also_on_cli(self) -> None:
+        test_toml_file = THIS_DIR / "test_src.toml"
+        fake_ctx = FakeContext()
+        fake_ctx.params["src"] = ("src",)
+        black.read_pyproject_toml(
+            fake_ctx, FakeParameter(), str(test_toml_file),
+        )
+        config = fake_ctx.default_map
+        self.assertEqual(config["verbose"], "1")
+        self.assertEqual(config["check"], "no")
+        self.assertEqual(config["diff"], "y")
+        self.assertEqual(config["color"], "True")
+        self.assertEqual(config["line_length"], "79")
+        self.assertEqual(config["target_version"], ["py36", "py37", "py38"])
+        self.assertEqual(config["exclude"], r"\.pyi?$")
+        self.assertEqual(config["include"], r"\.py?$")
```

Comment:
I'm not sure if these tests are needed here, since they are already in other functions. Removing them would simplify the test a lot.